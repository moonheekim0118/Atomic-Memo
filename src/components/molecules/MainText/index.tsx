import React from 'react';
import styled from 'styled-components';
import Textarea from '../../atoms/Textarea';
import Span from '../../atoms/Span';

interface Props {
  value: string;
  onChange: (e: React.MouseEvent) => void;
  error: boolean;
}

const MainText = ({ value, onChange, error }: Props) => {
  return (
    <StyledMainText>
      <Textarea value={value} onChange={onChange} size="main" />
      <Span color={error ? 'red' : 'black'}>{`${value.length}/500`}</Span>
    </StyledMainText>
  );
};

const StyledMainText = styled.div`
  display: flex;
  flex-direction: column;
  padding: 3em 0;
`;

export default MainText;
