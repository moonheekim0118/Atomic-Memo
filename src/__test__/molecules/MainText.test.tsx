import 'jsdom-global/register';
import React from 'react';
import MainText from '../../components/molecules/MainText';
import { mount } from 'enzyme';

describe('<MainText/>', () => {
  let container = null;
  const mockFn = jest.fn();

  it('renders correctly', () => {
    container = mount(<MainText value="" onChange={mockFn} error={false} />);
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
