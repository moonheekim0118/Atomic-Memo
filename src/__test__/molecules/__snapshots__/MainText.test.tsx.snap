// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainText/> matches snpashot 1`] = `"<div class=\\"MainText__StyledMainText-sc-17zmbb-0 dgfvXV\\"><textarea rows=\\"10\\" font-size=\\"1.2rem\\" placeholder=\\"Contents\\" class=\\"Textarea__StyledTextArea-o90l14-0 kGrAoo\\"></textarea><span color=\\"black\\" width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 jaanRG normal\\">0/500</span></div>"`;
